deploy/test/test_cases/TEST_ID_1.7_providing_large_decoded_variable_successfully.sh

```diff
@@ -20,7 +20,7 @@ echo "Verifying pod test_env has environment variable '$environment_variable_nam
 test_pod="$(get_pod_name "$APP_NAMESPACE_NAME" 'app=test-env')"
 actual_value=$($cli_with_timeout "exec $test_pod -- printenv | grep VARIABLE_WITH_BASE64_SECRET | cut -d= -f2")
 
-if [[ "$actual_value" == "$secret_value" ]]; then
+if [[ "$actual_value" == "$secret_value"* ]]; then
```

Comment:
So actual_value could be secret_value plus more? What would the "plus more" part be here?

Reply:
The "plus more" is accounting for base64 padding characters.

Base64 encoded data sometimes contains between 1 or 2 padding characters (=) at the end in order to reach a length that is a multiple of 4. 

In switching to byte slices, the decoded output was (incorrectly) leaving these padding characters in the resulting secret as null bytes. I added code to trim any null bytes from the output, which means this comparison now needs to allow for the fact there may have been a couple bytes trimmed off the end of the input data.